server/src/index.ts

```diff
@@ -20,18 +22,64 @@ const app: Express = express();
 app.use(cors(config.server.cors));
 app.use(express.json());
 
+// Apply access control middleware to all routes except health check
+app.use((req, res, next) => {
+  if (req.path === '/health') {
+    return next();
+  }
+  validateAccessCode(req, res, next);
+});
+
 const PORT = config.server.port;
 
 // Create HTTP server
 const server = createServer(app);
 
 // Initialize Socket.IO
 const io = new Server(server, {
-  cors: config.server.cors,
+  cors: {
+    origin: config.server.cors.origin,
+    methods: config.server.cors.methods,
+    credentials: config.server.cors.credentials,
+    allowedHeaders: config.server.cors.allowedHeaders
+  },
   allowEIO3: true,
   transports: ['polling', 'websocket']
 });
 
+// Add Socket.IO middleware for access code validation
+io.use((socket, next) => {
+  const accessCode = socket.handshake.headers['x-access-code'] as string;
+  
+  if (!accessCode) {
+    return next(new Error('Access code is required'));
+  }
+  
+  const validAccessCodes = process.env.VALID_ACCESS_CODES?.split(',') || [];
```

Comment:
please update the .env.example file as well to reflect all the new env variables